### PR TITLE
 #536289: [SXA] Add custom template for _jss scaffold_

### PR DESCRIPTION
**Pull request by matkovskyi**
- [ ] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)

## Description / Motivation

Added nextjs-sxa scaffold component action in initializer. 

## Testing Details

- [ ] Unit Test Added
- [ ] Manual Test/Other (Please elaborate)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by ambrauer — COMMENTED** [review]
@matkovskyi any reason not to just include this with the existing `nextjs-sxa` initializer add-on?

**Comment by matkovskyi** [comment]
> @matkovskyi any reason not to just include this with the existing `nextjs-sxa` initializer add-on?

Yes, you're right! I fixed it

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/scripts/templates/component-src.ts
Comment (sc-dawidrutkowski) [inline packages/create-sitecore-jss/src/templates/nextjs-sxa/scripts/templates/component-src.ts:1]: @matkovskyi That is not enough. We must scaffold the basic structure of the SXA component including:
- usage of `props.params.styles`
- usage of `props.params.RenderingIdentifier`
- usage of `export const Default` which stands for the default rendering variant
No need of creating two variants, take a look at what e.g.: the Promo component looks like.

**Review by sc-dawidrutkowski — CHANGES_REQUESTED** [review]

**Comment by illiakovalenko** [comment]
@matkovskyi Please, also update CHANGELOG.md file to include your PR. We started to update Changelog for each PR, so it makes easier for us to start release processes

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/scripts/templates/component-src.ts
Comment (matkovskyi) [reply packages/create-sitecore-jss/src/templates/nextjs-sxa/scripts/templates/component-src.ts:1]: fixed

**Review by sc-dawidrutkowski — APPROVED** [review]

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/scripts/scaffold-component.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/*
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/nextjs-sxa/scripts/scaffold-component.ts:1]: I'm not sure that I can see any difference of this file and nextjs/scripts/scaffold-component.ts that we have. Essentially this file is not needed

**Review by illiakovalenko — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ Our versioning strategy is as follows:
 - Major: may include breaking changes in core packages (e.g. major architectural changes, major features)
 
 ## Unreleased
+* `[templates/nextjs-sxa]` Add scaffold components action `jss scaffold {nameOfComponent}` ([#1420](https://github.com/Sitecore/jss/pull/1420))
```
Comment (illiakovalenko) [inline CHANGELOG.md:13]:
```suggestion
* `[templates/nextjs-sxa]` Add custom template for _jss scaffold_ ([#1420](https://github.com/Sitecore/jss/pull/1420))
```

**Review by illiakovalenko — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (illiakovalenko) [inline CHANGELOG.md:13]: @matkovskyi  It should be added under specific section: Bug Fix, Improvements - you can see example in a same file below.
In your case it's improvement

**Pull request merged**